src/bindings/python/src/pyopenvino/core/core.cpp

```diff
@@ -326,13 +326,38 @@ void regclass_Core(py::module m) {
     cls.def(
         "read_model",
         [](ov::Core& self, py::object model_path, py::object weights_path) {
-            std::string model_path_cpp{py::str(model_path)};
-            std::string weights_path_cpp{py::str(weights_path)};
-            py::gil_scoped_release release;
-            return self.read_model(model_path_cpp, weights_path_cpp);
+            std::stringstream _stream;
+            if (py::isinstance(model_path, pybind11::module::import("io").attr("BytesIO"))) {
```

Comment:
In the docstring `model` is defined as `str`, but it can also be `BytesIO` with this approach.

Reply:
updated docstrings